redfish-core/lib/account_service.hpp

```diff
@@ -2001,6 +2026,28 @@ inline void requestAccountServiceRoutes(App& app)
                 }
             }
 
+
+            // Don't PATCH any account which has a Restricted Role,
+            // except allow the Locked property.
+            // Do not even allow the service user to change these properties.
+            // Implementation note: Ideally this would get the user's RoleId
+            // but that would take an additional D-Bus operation.
+            if ((username == "service") &&
+                (newUserName || password || roleId || enabled))
+            {
+                BMCWEB_LOG_ERROR << "Attempt to PATCH user who has a Restricted Role";
+                messages::restrictedRole(asyncResp->res, "OemIBMServiceAgent");
+                return;
+            }
+
+            // Don't PATCH any account to have a Restricted Role.
```

Comment:
nitpick: that doesn't make sesne 
Don't PATCH any accounts that have ..."

Reply:
The comment should change to: Don't allow PATCHing in a Restricted Role
Per the log below.

Reply:
Will change to: Don't allow PATCHing a RoleId if the new role is Restricted.